### PR TITLE
move gen component out of makefile

### LINKED ISSUE #80: move gen component out of makefile
address comment from https://github.com/kurator-dev/kurator/pull/77#discussion_r977545985

### PR DESCRIPTION
**Pull request by zirain**
Signed-off-by: hejianpeng <hejianpeng2@huawei.com>

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #80 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: hack/init-gen-tools.sh [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+set -o nounset
+set -o pipefail
+
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)"
+
+source "${REPO_ROOT}/hack/util.sh"
+
+# install prerequisite tools
+if ! [ -x "$(command -v jb)" ]; then
+    util::install_tools github.com/jsonnet-bundler/jsonnet-bundler/cmd/jb latest
```
Comment (hzxuzhonghu) [inline hack/init-gen-tools.sh:13]: we should use a stable release

File: hack/init-gen-tools.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+fi
+if ! [ -x "$(command -v gojsontoyaml)" ]; then
+    util::install_tools github.com/brancz/gojsontoyaml latest
```
Comment (hzxuzhonghu) [inline hack/init-gen-tools.sh:16]: same here

File: hack/init-gen-tools.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+fi
+if ! [ -x "$(command -v jsonnet)" ]; then
+    util::install_tools github.com/google/go-jsonnet/cmd/jsonnet latest
```
Comment (hzxuzhonghu) [inline hack/init-gen-tools.sh:19]: same

File: hack/init-gen-tools.sh (hunk 1 above)
Comment (zirain) [reply hack/init-gen-tools.sh]: fixed

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**